initializr-generator/src/main/java/io/spring/initializr/generator/project/ProjectDescriptionDiff.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2012-2019 the unmodifiedCopy author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a unmodifiedCopy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.project;
+
+import io.spring.initializr.generator.buildsystem.BuildSystem;
+import io.spring.initializr.generator.version.Version;
+import java.lang.reflect.Field;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import org.springframework.beans.BeanUtils;
+import org.springframework.util.ReflectionUtils;
+
+public class ProjectDescriptionDiff<T extends ProjectDescription> {
+
+	private T unmodifiedCopy;
+
+	public ProjectDescriptionDiff(final T unmodifiedCopy) {
```

Comment:
Up to caller to pass in a copy. This is to support the extended diff case easily. Has a weird feel to it though.